Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/NewEventStatsAnalyzer.cs

```diff
@@ -0,0 +1,159 @@
+                   opt.Name.Equals("o", StringComparison.OrdinalIgnoreCase))
+                {
+                    outputFilePath = ParseSingletonPathOption(opt, outputFilePath);
+                }
+                else if (opt.Name.Equals("inputDir", StringComparison.OrdinalIgnoreCase) ||
+                     opt.Name.Equals("i", StringComparison.OrdinalIgnoreCase))
+                {
+                    inputDirPath = ParseSingletonPathOption(opt, inputDirPath);
+                }
+                else
+                {
+                    throw Error("Unknown option for event stats analysis: {0}", opt.Name);
+                }
+            }
+
+
+            if (string.IsNullOrEmpty(inputDirPath))
+            {
+                throw Error("/inputDir parameter is required");
+            }
+
+            if (string.IsNullOrEmpty(outputFilePath))
+            {
+                throw Error("/outputFile parameter is required");
+            }
+
+            return new NewEventStatsAnalyzer(GetAnalysisInput())
+            {
+                InputDirPath = inputDirPath,
+                OutputFilePath = outputFilePath
+            };
+        }
+
+        /// <summary>
+        /// Write the help message when the analyzer is invoked with the /help flag
+        /// </summary>
+        private static void WriteNewEventStatsHelp(HelpWriter writer)
+        {
+            writer.WriteBanner("New Event Stats Analyzer");
+            writer.WriteModeOption(nameof(AnalysisMode.NewEventStats), "Generates stats on the aggregate size and count of execution log events, but uses the RocksDB database as the source of truth");
+            writer.WriteOption("inputDir", "Required. The directory to read the RocksDB database from", shortName: "i");
+            writer.WriteOption("outputFile", "Required. The file where to write the results", shortName: "o");
+        }
+    }
+
+    /// <summary>
+    /// Analyzer used to get stats on events (count and total size)
+    /// </summary>
+    internal sealed class NewEventStatsAnalyzer : Analyzer
+    {
+        /// <summary>
+        /// Input directory path that contains the RocksDB sst files
+        /// </summary>
+        public string InputDirPath;
+
+        /// <summary>
+        /// Output file path where the results will be written to.
+        /// </summary>
+        public string OutputFilePath;
+
+        private Stopwatch m_stopWatch;
+
+        public NewEventStatsAnalyzer(AnalysisInput input) : base(input)
+        {
+            m_stopWatch = new Stopwatch();
+            m_stopWatch.Start();
+        }
+
+        /// <inheritdoc/>
+        public override int Analyze()
+        {
+            using (var dataStore = new XldbDataStore(storeDirectory: InputDirPath))
+            {
+                using (var outputStream = File.OpenWrite(OutputFilePath))
+                {
+                    using (var writer = new StreamWriter(outputStream))

```

Comment:
you don't need braces for nested `using` statements, i.e., you should write it like:
```csharp
using (var dataStore = new XldbDataStore(storeDirectory: InputDirPath))
using (var outputStream = File.OpenWrite(OutputFilePath))
using (var writer = new StreamWriter(outputStream))
{
    ...
}
```